File: 백준/Silver/14916. 거스름돈/거스름돈.py
n = int(input())
cnt = 0
while n > 0:
    if (n%5) % 2 == 0:
        k = n//5
        cnt += k
        n -= 5 * k
        cnt += n//2
        break
    else:
        n -= 2
        cnt += 1

if n < 0:
    print(-1)
else:
    print(cnt)

File: 백준/Bronze/5585. 거스름돈/거스름돈.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    static StringBuilder sb = new StringBuilder();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int balance = 1000 - n;
        int[] coin = {500, 100, 50, 10, 5, 1};
        int cnt = 0;

        for (int i = 0; i < 6; i++) {
            cnt += balance / coin[i];
            balance %= coin[i];
        }
        System.out.println(cnt);

    }
}
